sdk/python/lib/pulumi_policy/policy.py

```diff
@@ -0,0 +1,751 @@
+    StackValidationPolicy is a policy that validates a stack.
+    """
+
+    __validate: Optional[StackValidation]
+    """
+    Private field holding the optional validation callback.
+    """
+
+    def validate(self, args: StackValidationArgs, report_violation: ReportViolation) -> Optional[Awaitable]:
+        if not self.__validate:
+            raise NotImplementedError(f'`validate` must be overridden by policy "{self.name}"'
+                                      + ' since `validate` was not specified')
+
+        result = self.__validate(args, report_violation)
+        if result is not None and isawaitable(result):
+            return cast(Awaitable, result)
+
+        return None
+
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 validate: Optional[StackValidation] = None,
+                 enforcement_level: Optional[EnforcementLevel] = None) -> None:
+        """
+        :param str name: An ID for the policy. Must be unique within the current policy set.
+        :param str description: A brief description of the policy rule. e.g., "S3 buckets should have
+               default encryptionenabled."
+        :param Optional[StackValidation] validate: A callback function that validates if a stack violates a policy.
+        :param Optional[EnforcementLevel] enforcement_level: Indicates what to do on policy violation,
+               e.g., block deployment but allow override with proper permissions.
+        """
+        super().__init__(name, description, enforcement_level)
+
+        # If this instance isn't a subclass, then validate must be specified.
+        not_subclassed = type(self) is StackValidationPolicy # pylint: disable=unidiomatic-typecheck
+        if not_subclassed and not validate:
+            raise TypeError("Missing validate argument")
+
+        if validate:
+            if not callable(validate):
+                raise TypeError("Expected validate to be callable")
+
+        self.__validate = validate # type: ignore
+
+
+class _PolicyAnalyzerServicer(proto.AnalyzerServicer):
+    __policy_pack_name: str
+    __policy_pack_version: str
+    __policies: List[Policy]
+    __policy_pack_enforcement_level: EnforcementLevel
+
+    def Analyze(self, request, context):
+        diagnostics: List[proto.AnalyzeDiagnostic] = []
+        for policy in self.__policies:
+            enforcement_level = self._get_enforcement_level(policy)
+            if enforcement_level == EnforcementLevel.DISABLED or not isinstance(policy, ResourceValidationPolicy):
+                continue
+
+            report_violation = self._create_report_violation(diagnostics, policy.name,
+                                                             policy.description, enforcement_level)
+
+            # TODO[pulumi/pulumi-policy#208]: Deserialize properties
+            # TODO[pulumi/pulumi-policy#208]: Unknown checking proxy
+            props = json_format.MessageToDict(request.properties)
```

Comment:
What does happen to "unknowns" in the meantime?

Reply:
They'll have unknown sentinel values.

```go
const (
	// UnknownBoolValue is a sentinel indicating that a bool property's value is not known, because it depends on
	// a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownBoolValue = "1c4a061d-8072-4f0a-a4cb-0ff528b18fe7"
	// UnknownNumberValue is a sentinel indicating that a number property's value is not known, because it depends on
	// a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownNumberValue = "3eeb2bf0-c639-47a8-9e75-3b44932eb421"
	// UnknownStringValue is a sentinel indicating that a string property's value is not known, because it depends on
	// a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownStringValue = "04da6b54-80e4-46f7-96ec-b56ff0331ba9"
	// UnknownArrayValue is a sentinel indicating that an array property's value is not known, because it depends on
	// a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownArrayValue = "6a19a0b0-7e62-4c92-b797-7f8e31da9cc2"
	// UnknownAssetValue is a sentinel indicating that an asset property's value is not known, because it depends on
	// a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownAssetValue = "030794c1-ac77-496b-92df-f27374a8bd58"
	// UnknownArchiveValue is a sentinel indicating that an archive property's value is not known, because it depends
	// on a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownArchiveValue = "e48ece36-62e2-4504-bad9-02848725956a"
	// UnknownObjectValue is a sentinel indicating that an archive property's value is not known, because it depends
	// on a computation with values whose values themselves are not yet known (e.g., dependent upon an output property).
	UnknownObjectValue = "dd056dcd-154b-4c76-9bd3-c8f88648b5ff"
)
```

Planning to quickly address this, so it shouldn't be a problem for long. Aiming to have the change ready before the hackathon.